internal/client/client.go

```diff
@@ -661,3 +669,129 @@ func (c *RPCClient) CloseAddr(addr string) error {
 	}
 	return nil
 }
+
+type spanInfo struct {
+	name     string
+	dur      uint64
+	async    bool
+	children []spanInfo
+}
+
+func (si *spanInfo) calcDur() uint64 {
+	if si.dur == 0 {
+		for _, child := range si.children {
+			if child.async {
+				// TODO: Here we just skip the duration of async process, however there might be a sync point before a
+				// specified span, in which case we should take the max(main_routine_duration, async_span_duration).
+				// It's OK for now, since only pesist-log is marked as async, whose duration is typically smaller than
+				// commit-log.
+				continue
+			}
+			si.dur += child.calcDur()
+		}
+	}
+	return si.dur
+}
+
+func (si *spanInfo) addTo(parent opentracing.Span, start time.Time) time.Time {
+	if parent == nil {
+		return start
+	}
+	dur := si.calcDur()
+	if dur == 0 {
+		return start
+	}
+	end := start.Add(time.Duration(dur) * time.Nanosecond)
+	tracer := parent.Tracer()
+	span := tracer.StartSpan(si.name, opentracing.ChildOf(parent.Context()), opentracing.StartTime(start))
+	t := start
+	for _, child := range si.children {
+		t = child.addTo(span, t)
```

Comment:
Should we add the time of child to t before adding the next child (if not async)? Then, the start time of sequential spans won't be the same

Reply:
The start time of sequential spans won't be the same here. `spanInfo.addTo` return the end time of the span (if it's not async).

Reply:
Ah, sorry. I misread it.